Add BIN-2025-0003, OP_PAIRCOMMIT

Ademan opened the pull request:
Since PAIRCOMMIT has been assigned `BIP-442` does it make the most sense to just point the binana to the (currently not yet merged) BIP?

This is an alternative approach to #8 

I made an effort to follow existing conventions but I wasn't sure about a few things:

1. Self assignment of the BIP number->BIN number, is this OK or should I use BIN-2025-0003, or even steal #8 's BIN-2024-0006 ?  (BIP-119 which I copied referenced an older version scheme, is that still applicable here?)
2. PAIRCOMMIT has been assigned a BIP number but hasn't been merged yet, so this is pointing to something outside of the official BIPs repo, is this OK?
3. Wasn't sure what date to use for the Revision field, just used the current date, should it be the date of the referenced commit instead?

Finally, my ultimate goal is to add PAIRCOMMIT as inquisition heretical deployments, it appeared to me that this is a prerequisite to getting that upstreamed into [bitcoin-inquisition/bitcoin](https://github.com/bitcoin-inquisition/bitcoin) , right?

ajtowns commented:
BIN16-118 and BIN16-119's numbers are historical exceptions, and shouldn't be used as examples. Taking over BIN24-6 from #8 would make sense if the proposers of that are happy with it; if it's a divergent proposal, though, then it should get it's own number, I think BIN25-3 would be next?

Pointing to any reasonable permalink seems fine, particularly if a commit hash is included. If the permalink turns out not to be permanent, can always update it later, using the commit hash to ensure consistency.

Revision date should match the referenced document, but as long as it's not in the future, probably doesn't matter too much.

Inquisition heretical activations require a BINANA number, yeah.

moonsettler commented:
Feel free to take over, or tell me what to do with #8! I could update the PR to the latest against the BIP repo, but that is still under review and might receive feedback that results in a change.

Ademan commented:
I renamed this binana to BIN25-3 just to dodge any uncertainty about superceding #8.

I also updated the date to match the BIP date. The `PAIRCOMMIT` BIP hasn't been moving too fast lately but I'll make an effort to keep the pointer in this binana up to date.

Thanks for the feedback and guidance!

ajtowns commented:
Ugh, sorry about that. Tried doing the rebase myself to see if the new workflow was operative and messed up, and now github won't let me fix it. You should be able to rebase yourself and reopen the PR.
